storage/reader.go

```diff
@@ -195,8 +193,6 @@ func (o *ObjectHandle) NewMultiRangeDownloader(ctx context.Context) (mrd *MultiR
 	// span now if there is an error.
 	if err == nil {
 		r.ctx = ctx
-	} else {
-		trace.EndSpan(ctx, err)
 	}
```

Comment:
Why are we removing this?

Reply:
This reduces the risk of orphaned spans by adding the deferred pattern above. I tested with larger object sizes and through context propagation, the parent span still serves as an overarching span for the reader